### PR TITLE
cargBags_Nivaya improvements

### PR DESCRIPTION
**Pull request by ItsMattTrevino**
(Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.)

### What does this implement/fix? Explain your changes ###
It adds a third column to inventory management as well as better loot filters by way of additional bags for various specific items (like Herbalism related mats or items that allow the player to travel/teleport). More information regarding the custom bags that need to be created is provided in the newly created README for cargBags_Nivaya.

### Does this close any currently open issues? ###

Link to the issues here
    #60 

### Any relevant logs, error output, etc? ###

(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

### Any other comments? ###
Sorting using the arrows is a little buggy as pressing the arrows to sort under specific circumstances may result in the need to press them more than once to get them to go to where they need to go. Sorting from the last column (right) using the right arrow to get it back to the first column (left) no longer results in the bag being moved to that column.

### CODE REVIEW

**Comment by Gethe** [comment]
Looks good overall, but there are a few things. 1) RealUI uses spaces for indents, not tabs. There should be a setting in your editor. 2) Check the travis-ci report and fix all luacheck errors 3) Move the item ID tables outside the ClassifyItem function so they don't get recreated each time it's called.

**Comment by Gethe** [comment]
Also, you shouldn't need to call `GetItemInfo`, the `item` argument is a table and should already have all of that info. Check out https://github.com/RealUI/RealUI/blob/develop/cargBags_Nivaya/cargBags/base/implementation.lua#L334

**Comment by ItsMattTrevino** [comment]
You mentioned localization and I'm wondering if that will fail the item filtering when it checks for the existence of bags with certain names in order to sort to the appropriate locations in filters.lua.

File: cargBags_Nivaya/cargBags/mixins-add/layouts.lua [hunk 1]
```diff
@@ -50,7 +51,7 @@ function layouts.grid(self, columns, spacing, xOffset, yOffset)
         local xPos = (col - 1) * (width + spacing)
         local yPos = -1 * (row - 1) * (height + spacing)
 
-        button:ClearAllPoints()
+		button:ClearAllPoints()
```
Comment (MisatoTremor) [inline cargBags_Nivaya/cargBags/mixins-add/layouts.lua]: Seems to still use tabs, dunno how luacheck missed it

File: cargBags_Nivaya/cargBags_Nivaya.lua [hunk 2]
```diff
@@ -689,6 +694,13 @@ local function HandleSlash(msg)
         _G.cBnivCfg.BankCustomBags = not _G.cBnivCfg.BankCustomBags
         StatusMsg("Display of custom containers in the bank is now ", ". Reload your UI for this change to take effect!", _G.cBnivCfg.BankCustomBags, true, false)
 
+    -- /cbniv ? Item Name OR item number
+    -- Returns detailed information about the item in question
+	-- luacheck: globals GetItemInfo, ignore print
```
Comment (MisatoTremor) [inline cargBags_Nivaya/cargBags_Nivaya.lua]: Seems to still use tabs, dunno how luacheck missed it

File: cargBags_Nivaya/filters.lua [hunk 3]
```diff
@@ -12,6 +12,272 @@ local itemClass = ns.itemClass
 ns.existsBankBag = { Armor = true, Quest = true, TradeGoods = true, Consumables = true, BattlePet = true }
 ns.filterEnabled = { Armor = true, Quest = true, TradeGoods = true, Consumables = true, Junk = true, Stuff = true, ItemSets = true, BattlePet = true }
 
+-- Item IDs for classification purposes
+local itemIDs = {
+
+	-- Mechagon Tinkering
```
Comment (MisatoTremor) [inline cargBags_Nivaya/filters.lua]: Seems to still use tabs, dunno how luacheck missed it (following lines, too)

File: cargBags_Nivaya/filters.lua [hunk 4]
```diff
@@ -12,6 +12,272 @@ local itemClass = ns.itemClass
...
+local bagNames = {}
+if gl == "deDE" then
+-- German
+    bagNames.bagParts                  = "Handwerk: Teile"
+    bagNames.bagJewelcrafting          = "Handwerk: Juwelenschleifen"
+    bagNames.bagCloth                  = "Handwerk: Stoff"
+    bagNames.bagLeatherworking         = "Handwerk: Lederverarbeitung"
+    bagNames.bagMetalStone             = "Handwerkskunst: Metall & Stein"
+    bagNames.bagCooking                = "Handwerkskunst: Kochen"
+    bagNames.bagHerb                   = "Handelsfertigkeit: Herb"
```
Comment (MisatoTremor) [inline cargBags_Nivaya/filters.lua]: Should be: "Handwerk: Kräuter"

File: cargBags_Nivaya/filters.lua [hunk 5]
```diff
@@ -12,6 +12,272 @@ local itemClass = ns.itemClass
...
+}
+
+local bagNames = {}
+if gl == "deDE" then
+-- German
+    bagNames.bagParts                  = "Handwerk: Teile"
+    bagNames.bagJewelcrafting          = "Handwerk: Juwelenschleifen"
+    bagNames.bagCloth                  = "Handwerk: Stoff"
+    bagNames.bagLeatherworking         = "Handwerk: Lederverarbeitung"
+    bagNames.bagMetalStone             = "Handwerkskunst: Metall & Stein"
```
Comment (MisatoTremor) [inline cargBags_Nivaya/filters.lua]: Should be: "Handwerk: Metall & Stein"

File: cargBags_Nivaya/filters.lua [hunk 6]
```diff
@@ -12,6 +12,272 @@ local itemClass = ns.itemClass
...
+    bagNames.bagCooking                = "Handwerkskunst: Kochen"
```
Comment (MisatoTremor) [inline cargBags_Nivaya/filters.lua]: Should be: "Handwerk: Kochen"

File: cargBags_Nivaya/filters.lua [hunk 7]
```diff
@@ -12,6 +12,272 @@ local itemClass = ns.itemClass
...
+    bagNames.bagHerb                   = "Handelsfertigkeit: Herb"
+    bagNames.bagElemental              = "Handelsfertigkeit: Elementar"
```
Comment (MisatoTremor) [inline cargBags_Nivaya/filters.lua]: Should be: "Handwerk: Elementar"

File: cargBags_Nivaya/filters.lua [hunk 8]
```diff
@@ -12,6 +12,272 @@ local itemClass = ns.itemClass
...
+    bagNames.bagEnchanting             = "Handwerkskunst: Bezaubernd"
```
Comment (MisatoTremor) [inline cargBags_Nivaya/filters.lua]: Should be: "Handwerk: Verzauberkunst"

**Comment by MisatoTremor** [comment]
Otherwise this looks like a great addition / improvement! <3

**Comment by ItsMattTrevino** [comment]
Something tells me the other translations may need to be fixed up down the line - admittedly, I referred to Google translate for the translation work.

**Comment by MisatoTremor** [comment]
Maybe you could use Wowhead, they at least have a translation for every tradeskill. Thinking of this, some bags are named after their respective tradeskill and others by the material names. Dunno if it would be better to unify this.

**Comment by ItsMattTrevino** [comment]
> Maybe you could use Wowhead, they at least have a translation for every tradeskill. Thinking of this, some bags are named after their respective tradeskill and others by the material names. Dunno if it would be better to unify this.

I was originally going to do that but didn't want to overlap somebody who may have already been using the same name for another custom bag but, honestly, it's really what I should have done from the beginning.

**Pull request merged**